### ISSUE TITLE
"Failed to fetch" with self-signed 

### ISSUE THREAD

**Issue by ArthurAttout** [bitwarden/desktop#129]
I have a self-hosted environment with a self-signed SSL certificate (generated by the wizard). 

I set the server URL with `https://[ip_address]`.
If I try to login with an existing account, I get an error "Failed to fetch" with no further information. No container shows any attempt of connection, the error may then come from the desktop application itself. 

I manually installed the .crt file on the local machine, and restarted the computer. 
Has anyone expected this ? Or has a self-hosted environment with self-signed certificate that works ?

**Comment by kspearrin** [comment]
I recently tested this and it worked . Are you sure the certificate is properly trusted in the local cert store?

**Comment by ArthurAttout** [comment]
I took the certificate.crt under `bwdata\ssl\self\192.168.1.67` and installed it in the "personal" store. Should I use another one ?

**Comment by kspearrin** [comment]
Sounds right to me. You can go to View -> Developer Tools and look at console/network tabs for more info.

**Comment by ArthurAttout** [comment]
After checking the console, an error indeed showed up 

```
POST https://192.168.1.67/api/accounts/prelogin net::ERR_INSECURE_RESPONSE
(anonymous) @ vendor.js:97465
fetch @ api.service.ts:793
(anonymous) @ api.service.ts:830
(anonymous) @ i18n.service.ts:108
ZoneAwarePromise @ vendor.js:97298
__awaiter @ i18n.service.ts:108
send @ api.service.ts:797
(anonymous) @ api.service.ts:202
(anonymous) @ i18n.service.ts:108
ZoneAwarePromise @ vendor.js:97298
__awaiter @ i18n.service.ts:108
postPrelogin @ api.service.ts:201
(anonymous) @ auth.service.ts:206
(anonymous) @ twoFactorProviderType.ts:9
ZoneAwarePromise @ vendor.js:97298
__awaiter @ twoFactorProviderType.ts:9
makePreloginKey @ auth.service.ts:201
(anonymous) @ auth.service.ts:116
(anonymous) @ twoFactorProviderType.ts:9
ZoneAwarePromise @ vendor.js:97298
__awaiter @ twoFactorProviderType.ts:9
logIn @ auth.service.ts:114
(anonymous) @ login.component.ts:74
(anonymous) @ environment.component.ts:63
ZoneAwarePromise @ vendor.js:97298
__awaiter @ environment.component.ts:63
submit @ login.component.ts:56
(anonymous) @ login.component.html:1
handleEvent @ vendor.js:13904
handleEvent @ vendor.js:14636
dispatchEvent @ vendor.js:10316
(anonymous) @ vendor.js:12655
schedulerFn @ vendor.js:4703
SafeSubscriber.__tryOrUnsub @ vendor.js:37907
SafeSubscriber.next @ vendor.js:37854
Subscriber._next @ vendor.js:37794
Subscriber.next @ vendor.js:37758
Subject.next @ vendor.js:54447
EventEmitter.emit @ vendor.js:4671
NgForm.onSubmit @ vendor.js:44026
(anonymous) @ login.component.html:1
handleEvent @ vendor.js:13904
handleEvent @ vendor.js:14636
dispatchEvent @ vendor.js:10316
(anonymous) @ vendor.js:10941
(anonymous) @ vendor.js:50197
ZoneDelegate.invokeTask @ vendor.js:96832
onInvokeTask @ vendor.js:5096
ZoneDelegate.invokeTask @ vendor.js:96831
Zone.runTask @ vendor.js:96599
ZoneTask.invokeTask @ vendor.js:96906
invokeTask @ vendor.js:97947
globalZoneAwareCallback @ vendor.js:97973
```

I tried adding the certificate through Chrome's SSL certificate management settings, but no luck ..

**Comment by ArthurAttout** [comment]
I'll keep on trying to install the cert properly and keep the thread updated

**Comment by kspearrin** [comment]
What about another browser, like Edge or Firefox. Do they trust it?

**Comment by ArthurAttout** [comment]
When viewing the target IP via any browser, I still have warnings that the website is not secure (I had to add an exception for it) because it is self-signed. I'll try to remove the exceptions I added for it, until it gets 100% trusted, but I don't think that's possible. I'll still have warnings even after that

**Comment by ArthurAttout** [comment]
Ok, I managed to get it working. 

This may be a stupid mistake, but I'm not super familiar with SSL and some people may do the same mistake. 

When going through the prompts of the wizard when installing Bitwarden with Docker, I entered an **IP address** when it prompted me for the **domain name** of the Bitwarden instance. Neither chrome nor firefox seemed to handle it (or maybe it's an expected behavior, I don't know). 

After re-installing bitwarden with a domain name (a silly one, hardcoded in etc/hosts), the certificate could be installed on machines, the Desktop app works and Chrome is also happy about it. 

I still have issues with Firefox, that still complains that the certificate is self-signed, but it's unrelated. 

This issue may be closed. 
Thanks for the help !
